### DISCUSSION TITLE
Pumpkin Revenue generation

### DISCUSSION THREAD

**Discussion by Snowiiii** [Pumpkin-MC/Pumpkin#474] [General]
Hello, Pumpkin Developers, Contributors, and Users!

In this issue, I'd like to discuss ways to generate sustainable funding for the Pumpkin project, ensuring a user and developer-friendly experience.

### Project Costs
- Website Hosting
- Domain costs
- Developer Time
- Living expenses

To mention are the Plugin Marketplace we planed. This will require some processing power and a good amount- of storage.

### Charging Hosting companies
Charging Hosting companies who are providing pumpkin game servers. I my opinion, They Basically make money with Pumpkin so why not get a piece of the cake. There where also a idea of making an own Hosting company and providing pumpkin game servers. Having a larger fee we could advertise with much lower costs
### Making an own Hosting Company
I mean i always wanted a own hosting company, I don't know if this fits well into this issue because its more of an personal project. But when such a project will come it will be open source and have pumpkin game servers.
### Charging Server owners
I had also an idea of charging Server owners when for example they make over ... € a Month. This will make also bigger Servers (e.g. Hypixel) who are self hosting have to be charged. But im not sure if this is a good idea, im predicting a good amout of backlash.
## Enterprise support :eyes:
A great solution I think. Companies need to have enterprise support. Their worst nightmare is if their servers go down and they have no professional to fix it, which is often the case with open source. What do they do then? Send a random message on Discord or Reddit asking if someone can help, and maybe get ghosted? Or, even if they don't get ghosted, how can they trust that the other person actually knows the software well enough? See this video I recently discovered on this topic: https://www.youtube.com/watch?v=2ymwP6_Fu0U&t=1s
### Provide a "Enterprise" or "Advanced" version of pumpkin :-1: 
Basically a closed-source version of Pumpkin with more features as the standard one.
- You need to maintain more code
- You have no contributors
- You have to prevent leaks on the internet
- It's not open source (i love open source)

I don't like this option
### Plugin Marketplace
Like said above there where plans of making something like the spigot plugin shop/download page. Where free plugin but also paid plugins can be downloaded/bought. We couldn't take a share when a plugin is bought and give the rest to the plugin developer. I think spigot does this as well, Steam does it as well. Sounds like a really good option to me. But building such a Store needs lot of time.
### Selling own Plugin
Selling own Plugins in on our own Plugin Marketplace :exploding_head:. Good solution
### Website Ads
Putting Ads on our Website, Yeah lets also may sell all the user data to google.
Nah i don't like this
### Developer Courses
Isn't this like just pay-to-win documentation? I'm not sure. I mean, if it's a video conference, maybe not. But that would require a lot of time. We could just let users buy pre-recorded videos explaining the Pumpkin API, or static documentation pages. But I'd rather publish tutorials on my YouTube channel for free.
### Sponsored Content
Sounds like a good solution.

## Revenue Distribution
After covering essential project costs, any remaining funds will be distributed to active developers and contributors.

**Comment by piprett** [comment]
Your conclusion is also what my opinion is, focusing on hosting companies. They can have big savings by using performant softwares, and with fast enough boot time they can even shut down the servers and spin it up once someone connects. Companies have big wallets to spend on Minecraft, Minecraft players usually don't. I don't like any of the following:
- Enterprise edition
- Plugin marketplace
- Selling plugins
- Advertisements
- Courses

**Comment by Snowiiii** [comment]
> Your conclusion is also what my opinion is, focusing on hosting companies. They can have big savings by using performant softwares, and with fast enough boot time they can even shut down the servers and spin it up once someone connects. Companies have big wallets to spend on Minecraft, Minecraft players usually don't. I don't like any of the following:
> 
>     * Enterprise edition
> 
>     * Plugin marketplace
> 
>     * Selling plugins
> 
>     * Advertisements
> 
>     * Courses

Sounds reasonable. But i wonder why you dislike the idea of an Plugin marketplace?

**Comment by piprett** [comment]
> Sounds reasonable. But i wonder why you dislike the idea of an Plugin marketplace?

It feels out of scope for Pumpkin, and it's a big "investment" of time that could be spent on the server software. Modrinth does this, it distributes plugins (but not paid). They have full-time employees, we don't. Realistically there isn't much to earn on it, from our side. The plugin creators wants their revenue, and I honestly don't believe that many people are gonna buy things. If we make hosting companies pay, we get money regardless if they play with or without plugins.

**Comment by lukas0008** [comment]
> > Sounds reasonable. But i wonder why you dislike the idea of an Plugin marketplace?
> 
> It feels out of scope for Pumpkin, and it's a big "investment" of time that could be spent on the server software. Modrinth does this, it distributes plugins (but not paid). They have full-time employees, we don't. Realistically there isn't much to earn on it, from our side. The plugin creators wants their revenue, and I honestly don't believe that many people are gonna buy things. If we make hosting companies pay, we get money regardless if they play with or without plugins.


I think the plugin marketplace is something that might not generate a lot of revenue, but is important to make nonetheless. The only reason why making spigot servers got so popular, is because of sites that host plugins. Having an official solution, that we don't have to work around, in my opinion seems like a good idea.

**Comment by DaniD3v** [comment]
> It feels out of scope for Pumpkin, and it's a big "investment" of time that could be spent on the server software

I feel like a official plugin marketplace would also be a quality of life improvement for pumpkin users, not only a source of revenue.
Would you rather have to search 3 shady websites for plugins or simply go to the 1 official pumpkin store?

> Modrinth does this, it distributes plugins (but not paid). They have full-time employees, we don't.
Your point is definitely valid but I just wanted to mention that modrinth does way more than hosting plugins so this is not a fair comparison. They host resource packs, shader packs, modpacks, etc.

Also a pumpkin specific marketplace would have other benefits like immediately showing a user 
which permissions a plugin needs / which language it is written in etc. 

This would only be possible with our own plugin metadata format.

**Comment by piprett** [comment]
> Would you rather have to search 3 shady websites for plugins or simply go to the 1 official pumpkin store?

Modrinth is one trusted website

> Also a pumpkin specific marketplace would have other benefits like immediately showing a user
which permissions a plugin needs / which language it is written in etc. This would only be possible with our own plugin metadata format.

What stops Modrinth from doing this? They already have custom code for every loader to parse and validate. What language is used doesn't matter, it's compiled to WASM so you don't even get to edit it.

**Comment by DaniD3v** [comment]
> Modrinth is one trusted website

It doesn't support paid plugins tho. 
Selling plugins is definitely a thing devs will want to do.

Those will have to go somewhere else. 

---

Another plus point is that we could integrate the plugin store with our server. Maybe you can just install a plugin like you would with cargo:

`pumpkin install world_edit`

**Comment by McModknower** [comment]
You guys could also provide support contracts. You sell contracts where larger customers pay you regularly, and if they have bugs or reasonable feature requests you prioritize them, and if they pay enough you even provide 24/7 support for emergencies.

**Comment by Snowiiii** [comment]
> You guys could also provide support contracts. You sell contracts where larger customers pay you regularly, and if they have bugs or reasonable feature requests you prioritize them, and if they pay enough you even provide 24/7 support for emergencies.

Isn't this basically the Enterprise support i mentioned?

**Comment by McModknower** [comment]
> > You guys could also provide support contracts. You sell contracts where larger customers pay you regularly, and if they have bugs or reasonable feature requests you prioritize them, and if they pay enough you even provide 24/7 support for emergencies.
> 
> Isn't this basically the Enterprise support i mentioned?

I just thought at first that the Enterprise support section was a section that groups all the sections below it. But if you mean that as its own point, yep its the same.

**Comment by lokka30** [comment]
I think selling your own plugins and offering Patreon tiers would be the best way to monetize the project.

Making your own hosting company could work, but it's a very difficult operation, especially considering the amount of user support you'll need to provide for issues you haven't even caused.

Enterprise support is probably a good idea to offer, though I doubt it will make much revenue.

***

My reasoning against the other methods:

- Charging Hosting companies.
  - None of them will pay for it, since the license permits them not to. It's impossible to make the project libre and simultaneously force hosting companies to pay. It will be very difficult to enforce.
- ^^ Same reasoning against charging server owners. Most can't afford paying anyhow since they're already paying for hosting.
- Plugin marketplace.
  - There's already Modrinth - no need to reinvent the wheel. These platforms are very difficult to develop.

The other ideas mentioned in the original post have obvious faults so I won't talk about those unless you'd like my opinion on them.

**Comment by DataM0del** [comment]
@Snowiiii just don't be sponsored by amazon or betterhelp and that gets rid of 50% of the problem :pray:
